### ISSUE TITLE
CI: pycparser install failure on one travis job

### ISSUE THREAD

**Issue by pv** [numpy/numpy#15108]
One of the Travis CI jobs fails to "pip install pycparser":

https://travis-ci.org/numpy/numpy/jobs/625256277#L390
(see under `./tools/travis-before-install.sh`).

It's the "python: 3.7, amd64, PYTHONOPTIMIZE=2 BLAS=None LAPACK=None ATLAS=None NPY_BLAS_ORDER=mkl,blis,openblas,atlas,accelerate,blas NPY_LAPACK_ORDER=MKL,OPENBLAS,ATLAS,ACCELERATE,LAPACK USE_ASV=1" one.

The `travis-before-install.sh` script is missing `set -e` on the top, so the failure is silent. Apparently the CI job still passes, but unclear if its working as it should currently.

**Comment by mattip** [comment]
Later in the run we again call `$PIP install -r test_requirements.txt` from `tools/travis-test.sh` and in that one it claims pycparser is already installed. Not clear from the log what is going on, it is too verbose. Is there a build in the matrix where the call in `tools/travis-test.sh` is not redundant?

**Comment by WarrenWeckesser** [comment]
It is this closed issue: https://github.com/eliben/pycparser/issues/291

According to eliben, 

> It's a known issue. Pycparser doesn't work with optimized (-O) Python builds that strip out docstrings.

I don't know why the test ultimately works.  Does the use of `-O` change the second time `$PIP install -r test_requirements.txt` is run?

**Comment by WarrenWeckesser** [comment]
FWIW: I did a couple experiments in https://github.com/numpy/numpy/pull/15275:

First, I just changed `PYTHONOPTIMIZE` in .travis.yml from 2 to 1.  That fixed the problem, but presumably the optimization level is being set to 2 intentionally, so changing it to 1 isn't a correct fix.

Then, with `PYTHONOPTIMIZE` back to 2,  I added the line `pip install -v --isolated pycparser` before `pip install -r test_requirements.txt` in tools/travis-before-install.sh.  Using `--isolated` is supposed to tell pip to ignore environment variables.  That didn't work--the error log showed the same failure to install pycparser.  I might be misinterpreting the meaning of the `--isolated` argument.  (Eventually the tests still passed, as they do in the master branch.)
